common/shell-local/config.go

```diff
@@ -159,6 +162,25 @@ func Validate(config *Config) error {
 				fmt.Errorf("Bad script '%s': %s", path, err))
 		}
 	}
+
+	// Check for properly formatted go os types
+	supported_syslist := []string{"darwin", "freebsd", "linux", "openbsd", "solaris", "windows"}
+	if len(config.OnlyOn) > 0 {
+		for _, provided_os := range config.OnlyOn {
+			supported_os := false
```

Comment:
here, if `supported_syslist` was sorted, you could have used [sort.SearchStrings](https://golang.org/pkg/sort/#SearchStrings).

Also, changing var names to camlCase would be a good idea.
ex: `supported_syslist` to `supportedSyslist` 
This way packer's code would feel more uniform.

Reply:
I'll do that in a follow-up commit; I don't think we need a whole new review cycle for camelCase.